feature_details.md

```diff
@@ -0,0 +1,174 @@
+
+* When a navigation is initiated to this url, the UA caches a static pixel snapshot for the DOM sub-tree of each shared element. If there is any hierarchical relationship between these elements, they are captured as separate snapshots to allow them to animate independently. The parent element's snapshot excludes content for any embedded shared elements, drawing the background which was earlier occluded by the child element.
+
+* The feature supports a subset of [animatable properties](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_animated_properties) that can be interpolated during a transition. For each of those properties, the UA caches their start value to animate them during the transition. In addition to this metadata, we allow the developer to provide opaque contextual information (a unique identifier for the element, navigation trigger, scroll offset, site version etc.), referred to as propertyMap, about these elements to configure the transition.
+
+```
+document.documentTransition.setSameOriginTransitions(
+  {“foo.com/a.html” : [{element : document.html, propertyMap : {“id” : “root”}},
+                                 {element : document.getElementById(‘shared’), propertyMap : {“id” : “shared”}],
+   “foo.com/b.html” : [{element : document.html}]
+  });
+```
+
+* When the new Document loads, the developer can register an `EventListener` to be notified if a transition was initiated by the previous Document. The event, referred to as `handleTransition`, provides the metadata for elements captured from the previous Document and the previous URL, a detailed explanation is at [Referencing Elements](#referencing-elements). The aim of the `handleTransition` event is to allow the developer to define the animations for a transition using information from both Documents. For instance, a paired element transition may be dropped if the corresponding element in the new Document renders offscreen.
+
+* The `handleTransition` event provides a promise based API to declare the transition parameters asynchronously. The motivation to make this asynchronous is to allow the new Document to go through intermediate layouts before it is ready to be displayed. The first paint for the Document is deferred and the previous Document is displayed until this promise is fulfilled and the transition can be started.
+
+```
+document.documentTransition.addEventListener(“handleTransition”, e => {
+    e.initializeTransition(transitionFrom(e.url, e.elements));
+});
+
+async function transitionFrom(previousUrl, elements) {
+    // Wait for resources required for this transition to load.
+    await waitForLoad(previousUrl);
```

Comment:
What is this waiting for? Does that mean that while we're running script for this page, we're still executing visual presentation from the previous page?

Reply:
I'll add the explanation for this with comments in the doc itself but here is a quick clarification.

This is waiting for any async activity required for first paint of the new Document to finish. The most common would be fetching assets from the network (images, fonts, scripts). The developer can decide how long to wait before the transition is started and what the fallback should be if we hit a timeout. For example, if an image on the new Document couldn't be loaded. What should we do if this image was a part of a paired transition?